components/binding-usage-controller/docs/proposals/usage-kind.md

```diff
@@ -92,6 +92,99 @@ The ServiceBindingUsage **spec.usedBy.kind** field matches the name of the **Usa
 
 It is important to have an easy way to create a test for a new **UsageKind**. There should be a way to check that the proper RBAC role is set or that the **labelsPath** field is correct. The purpose is to design a solution that makes writing such acceptance tests as easy as possible.
 
+## UsageKind and ServiceBindingUsage finalizers
+
+Binding Usage Controller implementation cannot handle **ServiceBindingUsage** deletion if the **UsageKind** has been deleted. The idea is to have a finalizer in the **UsageKind** instance. The finalizer is removed when the **UsageKind** is not used by any **ServiceBindingUsage**. Binding Usage Controller requires the **UsageKind** to handle **ServiceBindingUsage** deletion. This is the reason we need to have a finalizer in the **ServiceBindingUsage**. The deletion scenario looks like:
```

Comment:
`The Binding Usage Controller implementation cannot delete the **ServiceBindingUsage** if the **UsageKind** has been deleted. The idea is to have a finalizer in the **UsageKind** instance. The finalizer is removed only when the **UsageKind** is not used by any **ServiceBindingUsage**. The Binding Usage Controller requires the **UsageKind** to delete the **ServiceBindingUsage**. For this reason, a finalizer is needed in the **ServiceBindingUsage**. The deletion scenario looks as follows:`